### ISSUE TITLE
[Documentation Clarity] torch.min/torch.max gradient behavior

### ISSUE THREAD

**Issue by Enigmatisms** [pytorch/pytorch#160273]
### 📚 The doc issue

`torch.min` and `torch.max` do not mention the gradient behavior of theirs, yet the gradient behavior can be found in the documentation of `torch.amin`, `torch.amax`:
> Both max/min and amax/amin evenly distribute gradients between equal values when there are multiple input elements with the same minimum or maximum value.

Which seems, **unfortunately**, to be wrong, or at least, partially wrong, see the following example:

```python
>>> # torch.min -> reduce over all dimensions
>>> a = torch.ones([5]).cuda()
>>> a.requires_grad=True
>>> min_val = torch.min(a)
>>> min_val.backward()
>>> a.grad
tensor([0.2000, 0.2000, 0.2000, 0.2000, 0.2000], device='cuda:0')
...
>>> # torch.min(input, dim=...) -> reduce over specified dimension
>>> a = torch.ones([5]).cuda()
>>> a.requires_grad=True
>>> min_val = torch.min(a, dim=0)
>>> min_val.values.backward()
>>> a.grad
tensor([1., 0., 0., 0., 0.], device='cuda:0')
```

It is obvious that the former one performs a `amin`-like BP while the latter one is more like indexing. My torch version is `'2.7.1+cu126'`, and I think it is the same with CPU devices.

Can you make it clearer? (see Suggest a potential alternative/fix) 

### Suggest a potential alternative/fix

We know that `torch.min` (and `max`) has three sets of behaviors:
- reduce over all dimensions (single tensor input), its gradient behavior is the same as `amin`. (Evenly distribute).
- reduce over one specified axis, its gradient behavior is like using an `argmin` and a index select or something. (Propagate only to the indexed element)
- perform elementwise min/max (nothing to say here).

So, add the corresponding explanation for the gradient behavior, if I am correct about this. I've wasted a whole day on this issue, since either your documentation is way too simple for me to over-trust it or I am foolish on experimenting with PyTorch, sorry if my tone is harsh, wish PyTorch all the best, thanks.

cc @svekars @sekyondaMeta @AlannaBurke @albanD

**Comment by Raman-RH** [comment]
looks like we should provide the gradient details for min/max

**Comment by ngimel** [comment]
This behavior is intended.

**Comment by Enigmatisms** [comment]
Thanks
